ambry-server/src/test/java/com.github.ambry.server/AmbryRequestsTest.java

```diff
@@ -927,6 +1043,97 @@ private void doCatchupStatusTest(PartitionId id, long acceptableLagInBytes, shor
     assertEquals("Catchup status not as expected", expectedIsCaughtUp, adminResponse.isCaughtUp());
   }
 
+  // ttlUpdateTest() helpers
+
+  /**
+   * Does a TTL update and checks for success if {@code expected} is {@link ServerErrorCode#No_Error}. Else, checks for
+   * failure with the code {@code expected}.
+   * @param correlationId the correlation ID to use in the request
+   * @param clientId the client ID to use in the request
+   * @param blobId the blob ID to use in the request
+   * @param expiresAtMs the expiry time (ms) to use in the request
+   * @param opTimeMs the operation time (ms) to use in the request
+   * @throws InterruptedException
+   * @throws IOException
+   * @throws MessageFormatException
+   */
+  private void doTtlUpdate(int correlationId, String clientId, BlobId blobId, long expiresAtMs, long opTimeMs,
+      ServerErrorCode expected) throws InterruptedException, IOException, MessageFormatException {
+    TtlUpdateRequest request = new TtlUpdateRequest(correlationId, clientId, blobId, expiresAtMs, opTimeMs);
+    sendAndVerifyOperationRequest(request, expected, true);
+    if (expected == ServerErrorCode.No_Error) {
+      verifyTtlUpdate(request.getBlobId(), expiresAtMs, opTimeMs, MockStorageManager.messageWriteSetReceived);
+    }
+  }
+
+  /**
+   * Exercises various failure paths for TTL updates
+   * @throws InterruptedException
+   * @throws IOException
+   */
+  private void miscTtlUpdateFailuresTest() throws InterruptedException, IOException {
+    PartitionId id = clusterMap.getWritablePartitionIds(MockClusterMap.DEFAULT_PARTITION_CLASS).get(0);
+    // store exceptions
+    for (StoreErrorCodes code : StoreErrorCodes.values()) {
+      MockStorageManager.storeException = new StoreException("expected", code);
+      ServerErrorCode expectedErrorCode = ErrorMapping.getStoreErrorMapping(code);
+      sendAndVerifyOperationRequest(RequestOrResponseType.TtlUpdateRequest, Collections.singletonList(id),
+          expectedErrorCode, true);
+      MockStorageManager.storeException = null;
+    }
+    // runtime exception
+    MockStorageManager.runtimeException = new RuntimeException("expected");
+    sendAndVerifyOperationRequest(RequestOrResponseType.TtlUpdateRequest, Collections.singletonList(id),
+        ServerErrorCode.Unknown_Error, true);
+    MockStorageManager.runtimeException = null;
+
+    // store is not started/is stopped/otherwise unavailable - Disk_Unavailable
+    storageManager.returnNullStore = true;
+    sendAndVerifyOperationRequest(RequestOrResponseType.TtlUpdateRequest, Collections.singletonList(id),
+        ServerErrorCode.Disk_Unavailable, false);
```

Comment:
I have a PR #988 where `Replica_Unavailable` will be returned instead of `Disk_Unavailable` in this case. Anyway, we can fix this later.